### ISSUE TITLE
Generate per-namespace views for errors tables

### ISSUE THREAD

**Issue by whd** [mozilla/bigquery-etl#360]
As `payload_bytes_error` sees increased usage, we will want to set up user-facing views for accessing the data. Engineering currently has access to these tables via the BQ console, but the plan is to deprecate that access by creating an authorized view that elides some fields. This would also have the property of making the errors data more widely accessible beyond engineering e.g. to re:dash.

Since we provision datasets per namespace (not pipeline family), it would probably be useful to partitions error views by namespace as well so that e.g. `org-mozilla-fenix` users see an errors view in the `org-mozilla-fenix` dataset that has only fenix errors. We could do this in the view by matching on `uri`, probably.

The work of generating and managing these views might be done here or in the infra repo, I'm not sure presently. The ops infra currently calls generate/publish_views as part of schemas deploys, so hooking the view generation for errors into that process seems reasonable.

**Comment by jklukas** [comment]
> The ops infra currently calls generate/publish_views as part of schemas deploys, so hooking the view generation for errors into that process seems reasonable.

Agreed, although these will need to be authorized views, correct? I have not set one up before, so not exactly sure what the process is. It looks like it can all be done via the python api, so this is probably fine.

Do we know what group we want to grant view access to?

**Comment by whd** [comment]
See https://cloud.google.com/bigquery/docs/authorized-views and https://cloud.google.com/bigquery/docs/share-access-views. I believe you need the source dataset to grant access to the dataset+table the view will be in, which the ops infra has permission to do. We recently tested this with https://github.com/mozilla-services/cloudops-infra/pull/1379/files#diff-28af500d7cdc48a0f9b9728d8e886d22R146 and it appears to work.

**Comment by relud** [comment]
also noteworthy is that the view has to be re-granted access any time it changes

**Comment by mreid-moz** [comment]
In general, we no longer need views to make errors data accessible, but it may still be worthwhile to split the errors data up per-namespace to make it easier to look at errors for one namespace at a time.

**Comment by whd** [comment]
A note to update this that errors data is once again (as of late 2021) not generally accessible and so this issue is still relevant.

**Comment by ANich** [comment]
Seems like we could do this via `sql_generator` for the `document_namespace`'s that we care about, querying the already authorized views? cc @akkomar @kik-kik

**Comment by whd** [comment]
https://github.com/mozilla-services/cloudops-infra/pull/6192
